python/vineyard/data/arrow.py

```diff
@@ -51,7 +50,16 @@ def buffer_builder(
 
 
 def as_arrow_buffer(blob: Blob):
-    buffer = blob.buffer
+    if isinstance(blob, Blob):
+        buffer = blob.buffer
+    else:
+        if (
+            blob.id == ObjectID("o8000000000000000")
+            or (int(blob.id) & int(ObjectID("o8000000000000000"))) == 0
+        ):
+            buffer = b''
+        else:
+            buffer = memoryview(blob)
```

Comment:
We do have `is_empty`, https://github.com/v6d-io/v6d/blob/main/python/core.cc#L461.

You could add the comparison with empty blob id there, rather than hard code in Python code.

And, return both `memoryview(b'')` to keep the same return type in the if-then-else branches.